ParProcCo/job_controller.py

```diff
@@ -0,0 +1,64 @@
+from datetime import timedelta
+from pathlib import Path
+from job_scheduler import JobScheduler
+
+
+class JobController:
+
+    def __init__(self, working_directory, cluster_output_dir, project, priority, cpus=16, timeout=timedelta(hours=2)):
+        """
+        JobController is used to coordinate cluster job submissions with JobScheduler
+        Args:
+            working_directory (pathlib.Path or str): working directory
+            cluster_output_dir (pathlib.Path or str): cluster output directory
+            project (str): project name
+            priority (str): name of queue to submit to
+            cpus (int, optional): Number of CPUs to request. Defaults to 16.
+            timeout (int, optional): Timeout for cluster jobs in minutes. Defaults to 180 (2 hours).
+        """
+
+        self.working_directory = Path(working_directory)
+        self.cluster_output_dir = Path(cluster_output_dir)
+        self.project = project
+        self.priority = priority
+        self.cpus = cpus
+        self.timeout = timeout
+        self.scheduler = None
+        self.data_chunker = None
+
+    def run(self, data_chunker_class, data_chunker_args, data_to_split, processing_script):
+        self.data_chunker = data_chunker_class(*data_chunker_args)
+        input_paths = self.data_chunker.chunk(self.working_directory, data_to_split)
```

Comment:
The use cases include splitting processing by the jobs so each processes a subset of the same file. Therefore each job must be informed which subset to use via a command line argument